client/components/AllProducts.js

```diff
@@ -2,32 +2,60 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { fetchProducts } from '../store/products';
 import { Link } from 'react-router-dom';
+import FilterInput from './FilterInput';
 
 class AllProducts extends React.Component {
-    constructor(props) {
-        super(props)
-    }
-    componentDidMount() {
-        this.props.fetchInitialData();
-    }
-    render() {
-        return (
-            <div>
-                <ul>
-                    {this.props.products && this.props.products.map(product =>
-                        (<li key={product.id}>
-                            <Link to={`/products/${product.id}`}>{product.title}</Link>
-                        </li>)
-                    )}
-                </ul>
-            </div>
-        )
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: '',
     }
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const value = event.target.value;
```

Comment:
small nit, you could do `inputValue = event.target.value` and then `this.setState({ inputValue })`